api/preprints/views.py

```diff
@@ -69,6 +69,30 @@
 from osf.metrics import PreprintDownload, PreprintView
 
 
+class RejectPreprintEditMixin:
+    '''Override method to reject modify requests for old preprint versions (except for withdrawal)'''
```

Comment:
Ah, there might be a miscommunication here, I don't think we can update withdrawn preprints ... but we can withdrawn old versions.